### PR TITLE
Added ability to restrict bag recording to specific topics (fixes #341)

### LINKED ISSUE #341: Add restrict_to_topics to BagRecorder

### PR DESCRIPTION
**Pull request by ChrisTimperley**

### CODE REVIEW

File: src/roswire/proxy/bag.py [hunk 1]
```diff
@@ -221,6 +226,8 @@ def start(self) -> None:
                     f'__name:={self.__bag_name}']
             if self.__exclude_topics:
                 args += ['-x', shlex.quote(self.__exclude_topics)]
+            if self.__restrict_to_topics:
+                args += ['-e', shlex.quote(self.__restrict_to_topics)]
```
Comment (afsafzal) [inline src/roswire/proxy/bag.py:230]: I don't think `shlex.quote` here is doing what we want it to do. Right now, let's say for topics `(.*)/move_group/(.*)` it is generating the command as `-e (.*)/move_group/(.*)`, while it should be `-e "(.*)/move_group/(.*)"` to work.

I'm not sure how's best to deal with this. I have the dummy solution of:
```
args += ['-e', '"{}"'.format(shlex.quote(self.__restrict_to_topics))]
```

File: src/roswire/proxy/bag.py [hunk 2]
```diff
@@ -221,6 +226,8 @@ def start(self) -> None:
                     f'__name:={self.__bag_name}']
```
Comment (afsafzal) [inline src/roswire/proxy/bag.py:226]: Having the option `-a` in the arguments overrides option `-e`, and that's why it's not working.

Comment (ChrisTimperley) [reply src/roswire/proxy/bag.py:226]: Ah. That makes sense. Thanks for fixing this! :+1:

**Comment by ChrisTimperley** [comment]
The CI is failing, but upon closer look, it seems that `mypy` is finding issues elsewhere in the code that it hasn't identified in the past. Given that, I'll go ahead and merge this PR, and fix those newly discovered issues on another branch.

**Pull request merged**